test/collection.jl

```diff
@@ -45,6 +45,39 @@ using CCDReduction: parse_name
     @test size(df) == (0, 0)
 end
 
+@testset "array-generators" begin
+    # setting initial data
+    dir = joinpath(@__DIR__, "data")
+    df = fitscollection(dir)
+    arr1 = map(x -> x, array(df))
+    arr2 = map(eachrow(df)) do row
+        getdata(FITS(row.path)[row.hdu])
+    end
+    @test arr1 == arr2
+end
+
+@testset "filename-generators" begin
+    # setting initial data
+    dir = joinpath(@__DIR__, "data")
+    df = fitscollection(dir)
+    arr1 = map(x -> x, filename(df))
+    arr2 = map(eachrow(df)) do row
+        row.path
+    end
+    @test arr1 == arr2
+end
+
+@testset "image-generators" begin
+    # setting initial data
+    dir = joinpath(@__DIR__, "data")
+    df = fitscollection(dir)
+    arr1 = map(x->x, images(df))
+    arr2 = map(eachrow(df)) do row
+        FITS(row.path)[row.hdu]
+    end
+    @test arr1 == arr2
+end
```

Comment:
I can't understand why this test fails, in my machine it shows the following stack:
```
image-generators: Test Failed at /home/siddharth/.julia/dev/CCDReduction.jl/test/collection.jl:78
  Expression: arr1 == arr2
   Evaluated: ImageHDU[File: /home/siddharth/.julia/dev/CCDReduction.jl/test/data/M35070V.fits
HDU: 1
Type: Image
Datatype: Int16
Datasize: (1059, 1059), File: /home/siddharth/.julia/dev/CCDReduction.jl/test/data/M6707HH.fits
HDU: 1
Type: Image
Datatype: Int16
Datasize: (1059, 1059)] == ImageHDU[File: /home/siddharth/.julia/dev/CCDReduction.jl/test/data/M35070V.fits
HDU: 1
Type: Image
Datatype: Int16
Datasize: (1059, 1059), File: /home/siddharth/.julia/dev/CCDReduction.jl/test/data/M6707HH.fits
HDU: 1
Type: Image
Datatype: Int16
Datasize: (1059, 1059)]
```
PS: I think the left side and right side are equal.

Reply:
`ImageHDU` is a mutable struct